man/systemd.exec.xml

```diff
@@ -225,6 +225,55 @@
         <xi:include href="system-only.xml" xpointer="singular"/></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>ProtectProc=</varname></term>
+
+        <listitem><para>Takes one of <literal>noaccess</literal>, <literal>invisible</literal>,
+        <literal>ptraceable</literal> or <literal>default</literal> (which it defaults to). When set, this
+        controls the <literal>hidepid=</literal> mount option of the <literal>procfs</literal> instance for
+        the unit that controls which directories with process metainformation
+        (<filename>/proc/<replaceable>PID</replaceable></filename>) are visible and accessible: when set to
+        <literal>noaccess</literal> the ability to access most of other users' process metadata in
+        <filename>/proc/</filename> is taken away for processes of the service. When set to
+        <literal>invisible</literal> processes owned by other users are hidden from
+        <filename>/proc/</filename>. If <literal>ptraceable</literal> all processes that cannot be
+        <function>ptrace()</function>'ed by a process are hidden to it. If <literal>default</literal> no
+        restrictions on <filename>/proc/</filename> access or visibility are made. For further details see
+        <ulink url="https://www.kernel.org/doc/html/latest/filesystems/proc.html#mount-options">The /proc
+        Filesystem</ulink>. It is generally recommended to run most system services with this option
+        enabled. This option is implemented via file system namespacing, and thus cannot be used with
+        services that shall be able to install mount points in the host file system hierarchy. It also cannot
+        be used for services that need to access metainformation about other users' processes. This option
+        implies <varname>MountAPIVFS=</varname>.</para>
+
+        <para>If the kernel doesn't support per-mount point <option>hidepid=</option> mount options this
+        setting remains without effect, and the unit's processes will be able to access and see other process
+        as if the option was not used.</para>
+
+        <xi:include href="system-only.xml" xpointer="singular"/></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>ProcOnly=</varname></term>
```

Comment:
Naming is hard... But I think this name is problematic. First, it's rather unclear from the name what it is about. Only what? And the second issue is that the kernel interface will most likely evolve. As mentioned even in the description of this option, subset=pid is not really usable, and it seems very likely that we'll get get a new variant soon that allows basic api files to be also visible, so that normal programs can work. And then the boolean ProcOnly= option will be even more misleading.

So... can we please change this to ProcSubset=, and allow it to take "pid" and "default" (or "all") as the value right now, so that we are forward-compatible with kernel changes?